lib/zeitwerk/gem_loader.rb

```diff
@@ -23,6 +23,7 @@ def initialize(root_file, warn_on_extra_files:)
       @lib                 = File.dirname(root_file)
       @warn_on_extra_files = warn_on_extra_files
 
+      ignore(root_file) if @tag.include?("-")
```

Comment:
This is a hack. We could try to detect this better with convention, or perhaps add a `ignore_root_file` config option.